### PR TITLE
feat(radiogroup, checkboxgroup): deprecate radiogroup and checkboxgroup rules

### PR DESCRIPTION
**Pull request by straker**
As per https://github.com/dequelabs/axe-core/issues/176#issuecomment-195444801, deprecating the rules as they cause too many false positives.

As per our discussion on deprecating rules:

* can only happen in major or minor releases
* should disable the rule in the JSON
* should add the `deprecated` tag

Linked issue: #176 
Closes: #1315

## Reviewer checks

**Required fields, to be filled out by PR reviewer(s)**
- [x] Follows the commit message policy, appropriate for next version
- [x] Has documentation updated, a DU ticket, or requires no documentation change
- [x] Includes new tests, or was unnecessary
- [x] Code is reviewed for security by: Stephen

### CODE REVIEW

**Comment by straker** [comment]
Should also get @dylanb or @WilcoFiers approval before merging.

**Comment by dylanb** [comment]
Please hold this until we have had a discussion about the best path for replacement

File: lib/rules/checkboxgroup.json [hunk 1]
```diff
@@ -1,11 +1,12 @@
...
 	"selector": "input[type=checkbox][name]",
-	"tags": ["cat.forms", "best-practice"],
+	"tags": ["cat.forms", "best-practice", "deprecated"],
 	"metadata": {
 		"description": "Ensures related <input type=\"checkbox\"> elements have a group and that the group designation is consistent",
 		"help": "Checkbox inputs with the same name attribute value must be part of a group"
 	},
+	"enabled": false,
 	"all": [],
 	"any": ["group-labelledby", "fieldset"],
```
Comment (WilcoFiers) [inline lib/rules/checkboxgroup.json]: I created an issue, because these two checks will also have to be deprecated. That has to be in a different PR though: #1733

**Comment by mohanraj-r** [comment]
Are there suggested replacement rules to use in place of the deprecated ones ?
